radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/store/berkeley/BerkeleyTransactionsByIdStore.java

```diff
@@ -0,0 +1,283 @@
+import com.radixdlt.utils.RadixConstants;
+import com.radixdlt.utils.UInt256;
+import com.radixdlt.utils.UInt384;
+import com.sleepycat.je.Database;
+import com.sleepycat.je.DatabaseConfig;
+import com.sleepycat.je.DatabaseEntry;
+import com.sleepycat.je.Transaction;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.google.common.primitives.UnsignedBytes.lexicographicalComparator;
+import static com.sleepycat.je.LockMode.DEFAULT;
+import static com.sleepycat.je.OperationStatus.SUCCESS;
+
+public final class BerkeleyTransactionsByIdStore implements BerkeleyAdditionalStore {
+	private static final String TXN_ID_DB_NAME = "radix.transactions_by_id_db";
+	private Database txnIdDatabase; // Txns by AID; Append-only
+	private final AtomicReference<Instant> timestamp = new AtomicReference<>();
+	private final Addressing addressing;
+	private final Provider<RadixEngine<LedgerAndBFTProof>> radixEngineProvider;
+
+	@Inject
+	public BerkeleyTransactionsByIdStore(Provider<RadixEngine<LedgerAndBFTProof>> radixEngineProvider, Addressing addressing) {
+		// TODO: Fix this when we move AdditionalStore to be a RadixEngine construct rather than Berkeley construct
+		this.radixEngineProvider = radixEngineProvider;
+		this.addressing = addressing;
+	}
+
+	@Override
+	public void open(DatabaseEnvironment dbEnv) {
+		txnIdDatabase = dbEnv.getEnvironment().openDatabase(null, TXN_ID_DB_NAME, new DatabaseConfig()
+			.setAllowCreate(true)
+			.setTransactional(true)
+			.setKeyPrefixing(true)
+			.setBtreeComparator(lexicographicalComparator())
+		);
+	}
+
+	public boolean contains(AID aid) {
+		var key = new DatabaseEntry(aid.getBytes());
+		return SUCCESS == txnIdDatabase.get(null, key, null, DEFAULT);
+	}
+
+	public Optional<JSONObject> getTransactionJSON(AID aid) {
+		var key = new DatabaseEntry(aid.getBytes());
+		var value = new DatabaseEntry();
+
+		if (txnIdDatabase.get(null, key, value, DEFAULT) == SUCCESS) {
+			return Optional.of(new JSONObject(new String(value.getData(), StandardCharsets.UTF_8)));
+		}
+
+		return Optional.empty();
+	}
+
+	@Override
+	public void close() {
+		if (txnIdDatabase != null) {
+			txnIdDatabase.close();
+		}
+	}
+
+	private JSONObject mapToJSON(
+		Optional<TwoActorEntry> maybeEntry,
+		Function<REAddr, String> addrToRri,
+		BiFunction<ECPublicKey, UInt384, UInt384> computeStakeFromOwnership
+	) {
+		if (maybeEntry.isEmpty()) {
+			return new JSONObject().put("type", "Other");
+		}
+
+		var entry = maybeEntry.get();
+		var amtByteArray = entry.amount().toByteArray();
+		var amt = UInt256.from(amtByteArray);
+		var from = entry.from();
+		var to = entry.to();
+		var result = new JSONObject();
+		final ActionType type;
```

Comment:
I think that uninitialized variable is not the best approach. Perhaps more consistent would be to add "type" directly into `result` once action type is known. This also will make `Other` type handling consistent with remaining types.